pkg/actors/actors.go

```diff
@@ -1202,8 +1201,12 @@ func (a *actorsRuntime) CreateReminder(ctx context.Context, req *CreateReminderR
 		return err
 	}
 
-	a.activeRemindersLock.Lock()
-	defer a.activeRemindersLock.Unlock()
+	if !a.waitForEvaluationChan() {
```

Comment:
This was moved to before the lock because internally it uses a lock too and could otherwise cause a deadlock. Same in a couple more spots.